app/models/concerns/credit_card_validator.rb

```diff
@@ -0,0 +1,83 @@
+class CreditCardValidator < ActiveModel::EachValidator
```

Comment:
Fantastic work! Only thing of concern is that some of the business logic here (like determining the franchise) looks like it may be useful for contexts separate from validating the credit card. It's not a big deal right now, but something to keep in mind if there's some future work about using the franchise somewhere else.